### PR TITLE
Use mirrors to download Node.js binaries to escape sporadic 404 errors

### PR DESCRIPTION
**Pull request by ananzh**
Issue Resolved:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3618

 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Multiple DataSource] Refactor test connection to support SigV4 auth type ([#3456](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3456))
 - [Darwin] Add support for Darwin for running OpenSearch snapshots with `yarn opensearch snapshot` ([#3537](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3537))
 - [Vis Builder] Add metric to metric, bucket to bucket aggregation persistence ([#3495](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3495))
+- Fix nodejs download 404 error by switching to mirrors ([#3619](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3619))
```
[the remark this answers is not in the document]
Comment (ananzh) [reply CHANGELOG.md]: done

**Review by AMoo-Miki — APPROVED** [review]

**Comment by kavilla** [comment]
@peterzhuamazon for builds there is nothing here that would prevent us changing the URLs here right? Like some allowlist of links that the build process can pull from?

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon for builds there is nothing here that would prevent us changing the URLs here right? Like some allowlist of links that the build process can pull from?

It is not related to build so you can change per your need.
It is not pulling from ci.opensearch.org or artifacts.opensearch.org anyway.

Thanks.

**Review by abbyhu2000 — APPROVED** [review]

**Comment by joshuarrrr** [comment]
@ananzh We should backport to 2.x, right?

**Pull request merged**